frontend/src/pages/PipelineDetails.tsx

```diff
@@ -186,9 +194,38 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
     return this.load();
   }
 
+  private buildBreadCumb(
+    breadcrumbs: Array<{ displayName: string; href: string }>,
+    origin: origin,
+    runId: string,
+    experiment?: ApiExperiment,
+  ) {
+    // Build the breadcrumbs, by adding experiment and run names
+    if (experiment) {
+      breadcrumbs.push(
+        { displayName: 'Experiments', href: RoutePage.EXPERIMENTS },
+        {
+          displayName: experiment.name!,
+          href: RoutePage.EXPERIMENT_DETAILS.replace(
+            ':' + RouteParams.experimentId,
+            experiment.id!,
+          ),
+        },
+      );
+    } else {
+      breadcrumbs.push({ displayName: 'All runs', href: RoutePage.RUNS });
+    }
+    breadcrumbs.push({
+      displayName: origin.isRecurring ? origin.recurringRun!.name! : origin.run!.run?.name!,
+      href: RoutePage.RUN_DETAILS.replace(':' + RouteParams.runId, runId),
```

Comment:
I think this href will need to generate different URL based on whether it is recurring run or run. Right? RecurringRun doesn't have a run ID.